dashboard/test/controllers/data_docs_controller_test.rb

```diff
@@ -22,4 +27,24 @@ class DataDocsControllerTest < ActionController::TestCase
   test_user_gets_response_for :create, params: -> {@test_params}, user: :student, response: :forbidden
   test_user_gets_response_for :create, params: -> {@test_params}, user: :teacher, response: :forbidden
   test_user_gets_response_for :create, params: -> {@test_params}, user: :levelbuilder, response: :success
+
+  # TODO [meg] : all should be able to see `show` page, once launched
+  # for now, only levelbuilder can see it
+  test_user_gets_response_for :show, params: -> {{key: @data_doc_key}}, user: nil, response: :redirect
+  test_user_gets_response_for :show, params: -> {{key: @data_doc_key}}, user: :student, response: :forbidden
+  test_user_gets_response_for :show, params: -> {{key: @data_doc_key}}, user: :teacher, response: :forbidden
+  test_user_gets_response_for :show, params: -> {{key: @data_doc_key}}, user: :levelbuilder, response: :success
+
+  test 'creating a new data doc redirects to show page with key in URL' do
+    sign_in @levelbuilder
+    new_key = "doc_key"
+    get :create, params: {key: new_key}
+    assert_match %r{/data_docs/#{new_key}$}, @response.headers['Location']
```

Comment:
great test! one nit: rails provides a... railsier way to do this assertion: https://github.com/code-dot-org/code-dot-org/blob/f55ce24b2790f1babb48960810352a85f7d9e8ae/dashboard/test/controllers/followers_controller_test.rb#L31

that's what I'd recommend since it seems less fragile. if that seems too abstract, this could be a possible compromise: https://github.com/code-dot-org/code-dot-org/blob/cee622042b6eba2cd40fd396d1e85c5c300392b2/dashboard/test/controllers/projects_controller_test.rb#L360

Reply:
Oh, that's cool! Will do that 👍